### PR TITLE
feat: Multi-backend tracking, submit/benchmark CLIs, FSDP2 default

### PR DESCRIPTION
**Pull request by saforem2**
## Summary

- **Multi-backend experiment tracking**: New `Tracker` system with pluggable backends (wandb, MLflow, CSV) and unified `setup_tracker()` API. MLflow backend supports X-API-Key auth, dotenv loading, and artifact uploads.
- **New CLI commands**: `ezpz submit` for PBS/SLURM job submission and `ezpz benchmark` for orchestrated multi-example benchmark runs with progress, ETA, and summary reports.
- **FSDP2 as default**: `wrap_model()` now defaults to `fully_shard` (FSDP2) instead of DDP, with new `--fsdp-sharding-strategy` CLI arg and `reshard_after_forward` support.
- **Prefix-grouped History metrics**: Metrics logged with prefixed keys (e.g. `train/loss`, `eval/acc`) are now stored in separate groups internally. Each group gets its own xarray Dataset dimension — no more NaN padding between train and eval metrics. Plots show correct per-group x-axes with `train/loss`, `eval/loss` titles. `finalize()` saves one dataset per group and returns `dict[str, xr.Dataset]` when multiple groups exist.
- **Configurable log formatting**: Env-driven prefix styles (`EZPZ_LOG_PREFIX_STYLE`), timestamp visibility fixes, and new color options.
- **History + Tracker integration**: `History` now manages `Tracker` lifecycle internally for a single-object API. Tracker guide merged into the experiment tracking page. JSONL symlinks, output dir co-location, and finalize summaries.
- **MLflow system metrics**: `mlflow.enable_system_metrics_logging()` now enabled unconditionally (was gated on `MLFLOW_TRACKING_TOKEN`).
- **xccl split_group workaround**: Don't pass `device_id` for XPU/xccl backends in `init_process_group` to avoid PyTorch nightly regression in `DeviceMesh._unflatten` (pytorch/pytorch#178236).
- **Expanded test coverage**: 532 tests (up from ~200), new suites for PBS, tracker, log config, multirank, submit, TP, and History grouping.
- **Documentation overhaul**: Merged tracker.md into history.md, renamed reference.md to "End-to-End Walkthrough", added distributed training guide, 3 new recipes (data loading, checkpointing, gradient accumulation), updated index/quickstart features and cross-links.

## Test plan

- [x] Full test suite passes: 532 passed, 11 skipped
- [x] No broken doc links (verified with link checker)
- [x] Backward compatible: unprefixed metrics still return flat Dataset from finalize()

### CODE REVIEW

File: src/ezpz/examples/run_all.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    scheduler = "local"
+    job_id = str(os.getpid())
+    num_nodes = 1
+
+    if os.environ.get("PBS_JOBID"):
+        scheduler = "PBS"
+        job_id = os.environ["PBS_JOBID"]
+        nodefile = os.environ.get("PBS_NODEFILE", "")
+        if nodefile and Path(nodefile).exists():
+            num_nodes = len(Path(nodefile).read_text().splitlines())
```
[the remark this answers is not in the document]
Comment (saforem2) [reply src/ezpz/examples/run_all.py]: where in the world did you get that information ?

this is **absolutely** false, the `PBS_NODEFILE` contains one line per node; literally every single reference I could find says this explicitly: 

1. [Example Job Scripts - ALCF User Guides](https://docs.alcf.anl.gov/running-jobs/example-job-scripts/#hardware-threads_1)
2. [PBS Command: qsub](https://www.jlab.org/hpc/PBS/qsub.html)
3. [qsub - How does PBS\_NODEFILE work in pbs? - Stack Overflow](https://stackoverflow.com/questions/30742817/how-does-pbs-nodefile-work-in-pbs)
